cmd/gateway-gcs.go

```diff
@@ -294,8 +282,48 @@ func (l *gcsGateway) ListBuckets() ([]BucketInfo, error) {
 	return b, nil
 }
 
-// DeleteBucket delete a bucket on GCS
+// DeleteBucket delete a bucket on GCS.
 func (l *gcsGateway) DeleteBucket(bucket string) error {
+	itObject := l.client.Bucket(bucket).Objects(l.ctx, &storage.Query{Delimiter: slashSeparator, Versions: false})
+	// We list the bucket and if we find any objects we return BucketNotEmpty error. If we
+	// find only "minio.sys.temp/" then we remove it before deleting the bucket.
+	gcsMinioPathFound := false
+	nonGCSMinioPathFound := false
+	for {
```

Comment:
@balamurugana again, don't create a new function because it can be unit testable. Create a new function because it helps readability.
By the signature of the function itself you can notice that it is not intuitive:
```
func checkBucketEmpty(objIter ObjectIterator) error {
```

Also @abperiasamy said calls like this which needs to talk to backend cloud would be tested in functional tests and not unit tests.